IA-1320 Test cases for terra-jupyter-bioconductor

Qi77Qi opened the pull request:
…(#1075)

* Add the latest terra-jupyter-r:0.03 to automation testing.

* Add TODO list for test cases for terra-jupyter-bioconductor

* Add test cases for Bioconductor

This adds test cases for terra-jupyter-bioconductor

* Typo

\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

nturaga committed:
IA-1320 Test cases for terra-jupyter-bioconductor [Work in Progress] (#1075)

* Add the latest terra-jupyter-r:0.03 to automation testing.

* Add TODO list for test cases for terra-jupyter-bioconductor

* Add test cases for Bioconductor

This adds test cases for terra-jupyter-bioconductor

* Typo

nturaga committed:
IA-1320 Test cases for terra-jupyter-bioconductor [Work in Progress] (#1075)

* Add the latest terra-jupyter-r:0.03 to automation testing.

* Add TODO list for test cases for terra-jupyter-bioconductor

* Add test cases for Bioconductor

This adds test cases for terra-jupyter-bioconductor

* Typo

akarukappadath reviewed (changes requested):
I think that the `NotebookBioconductorKernelSpec` needs to be added to `LeonardoSuite` to be run with all the other autotests

akarukappadath reviewed (commented):
You'll also have to add the `bioconductorImageUrl` to `LeonardoConfig` - I think that'll help fix the compilation errors in the jenkins tests

akarukappadath committed:
config changes

akarukappadath committed:
compilation error fixes

akarukappadath committed:
compilation error fixes

jdcanas commented on automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookBioconductorKernelSpec.scala:
```diff
@@ -0,0 +1,299 @@
+    "should have GenomicFeatures automatically installed" in { clusterFixture =>
+      withWebDriver { implicit driver =>
+        withNewNotebook(clusterFixture.cluster, RKernel) { notebookPage =>
+          notebookPage.executeCell(""""GenomicFeatures" %in% installed.packages()""") shouldBe Some("TRUE")
```
these tests are fairly duplicative, consider creating a seq with the packages and using a for each, similar to https://github.com/DataBiosphere/leonardo/blob/develop/http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutesSpec.scala#L354

jdcanas commented on automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookBioconductorKernelSpec.scala:
```diff
@@ -0,0 +1,299 @@
+
+
+    "should be able to install packages that depend on libXML" in { clusterFixture =>
+      withWebDriver { implicit driver =>
```
possible reduction of repetition as above with slightly different structure

jdcanas commented on automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookBioconductorKernelSpec.scala:
```diff
@@ -0,0 +1,299 @@
+  override val jupyterDockerImage: Option[String] = Some(LeonardoConfig.Leonardo.bioconductorImageUrl)
+  "NotebookBioconductorKernelSpec" - {
+
+    "should use Bioconductor version 3.9" in { clusterFixture =>
```
consider parameterizing the version, as it will likely change.

jdcanas reviewed (approved):
I'd like to see the test cases generalized to avoid repetition, but looks good besides the test failures. They all seem to be failing due to `installation path not writeable`.

akarukappadath committed:
new bioc image

nturaga committed:
Modify test cases to "NOT" check for output after installing package.

This fails because of updates needed in future versions of dependencies. BiocManager::install will ask to update old packages as soon as there is a newer version, so the output changes for the packages being installed.

akarukappadath committed:
resolve conflicts

rtitle reviewed (approved):
Nice

rtitle commented on automation/src/test/resources/reference.conf:
```diff
@@ -5,4 +5,4 @@ leonardo.gatkImageUrl = "us.gcr.io/broad-dsp-gcr-public/terra-jupyter-gatk:0.0.4
 leonardo.oldWelderImage = "us.gcr.io/broad-dsp-gcr-public/welder-server:6a783a5"
 leonardo.currentWelderImage = "us.gcr.io/broad-dsp-gcr-public/welder-server:60e28bc" # This needs to be exactly the same in http/src/main/resources/reference.conf
-leonardo.bioconductor = "us.gcr.io/broad-dsp-gcr-public/terra-jupyter-bioconductor:0.0.4"
+leonardo.bioconductorImageUrl = "us.gcr.io/bioconductor-anvil/terra-jupyter-bioconductor@sha256:6bd0e493d38a5061f7621c0eb0fe383c3df1beb3432f93a4b570b9bcc59f129f"
```
Just curious why we're using a sha instead of a tag..

Qi77Qi replied:
I think this is just for Nitesh's testing purpose. It needs to be updated with https://github.com/DataBiosphere/leonardo/pull/1155

Qi77Qi committed:
update versions

Qi77Qi commented:
tests all passed...so I'm going to merge this
